testdata/reflect.go

```diff
@@ -580,3 +586,38 @@ func randuint32() uint32 {
 	xorshift32State = xorshift32(xorshift32State)
 	return xorshift32State
 }
+
+type U8 uint8
+
+func (v U8) Encode(buffer *bytes.Buffer) {
+	buffer.Write([]byte{0x00, 0x08})
+}
+
+// reflect was having trouble with tracking the exported flag when the item didn't fit in a pointer to when it did
+type TestU8 struct {
+	A U8
+	B U8
+	C U8
+	D U8
+	E U8 // fails if this additional field is added to the struct
+}
+
+//go:noinline
+func encode(value interface{}, buffer *bytes.Buffer) {
+	t := reflect.TypeOf(value)
+	v := reflect.ValueOf(value)
+
+	for i := 0; i < v.NumField(); i++ {
+		field := v.Field(i)
+		if t.Field(i).IsExported() && field.Kind() == reflect.Uint8 {
+			(field.Interface().(U8)).Encode(buffer)
+		}
+	}
+}
+
+func readBuffer(i int) int {
+	buffer := bytes.Buffer{}
+	t := TestU8{}
+	encode(t, &buffer)
+	return int(buffer.Bytes()[i])
+}
```

Comment:
I can reproduce the issue with the following code:

```
package main

import "reflect"

func main() {
        v := reflect.ValueOf(struct {
                A uintptr
                B uintptr
        }{2, 3})
        println(v.Field(0).Interface().(uintptr))
}
```

However, perhaps a test fix is the following:

```diff
diff --git a/testdata/reflect.go b/testdata/reflect.go
index 06bec575..47359e66 100644
--- a/testdata/reflect.go
+++ b/testdata/reflect.go
@@ -129,6 +129,7 @@ func main() {
                &linkedList{
                        foo: 42,
                },
+               struct{ A, B uintptr }{2, 3},
                // interfaces
                []interface{}{3, "str", -4 + 2.5i},
        } {
```
Without this fix, you can see there is a difference in output: TinyGo prints `caninterface=false` while Go doesn't. That's another way to verify this fix. (I haven't tested this PR but I assume it results in the same test output as `go run ./testdata/reflect.go`).